### PR TITLE
Rotation for Resculpaptable

### PR DESCRIPTION
**Pull request by pether-pg**
Adding rotation button to rotate resculptable art.

### CODE REVIEW

File: Resculpt/Resculptable.cs [hunk 1]
```diff
@@ -90,6 +90,22 @@ public sealed class Resculptable : KMonoBehaviour {
...
+		/// </summary>
+		private void OnRotateClicked()
+		{
+			Rotatable rotatable = this.gameObject.GetComponent<Rotatable>();
+			if (rotatable != null) rotatable.Rotate();
+
+			// Buildings with even width values jump one tile when rotating and must be moved back
+			BuildingDef def = this.gameObject.GetComponent<Building>()?.Def;
+			if (def != null && def.WidthInCells % 2 == 0)
+				this.transform.position += rotatable.GetOrientation() != Orientation.Neutral ? new UnityEngine.Vector3(1, 0, 0)
```
Comment (peterhaneve) [inline Resculpt/Resculptable.cs:105]: This statement could throw if `rotatable == null`, perhaps this should be included in the `if (rotatable != null)` block

File: Resculpt/Resculptable.cs [hunk 2]
```diff
@@ -90,6 +90,22 @@ public sealed class Resculptable : KMonoBehaviour {
...
+		/// <summary>
+		/// Triggered when the user requests a rotation of the decor item.
+		/// </summary>
+		private void OnRotateClicked()
+		{
+			Rotatable rotatable = this.gameObject.GetComponent<Rotatable>();
+			if (rotatable != null) rotatable.Rotate();
+
+			// Buildings with even width values jump one tile when rotating and must be moved back
+			BuildingDef def = this.gameObject.GetComponent<Building>()?.Def;
```
Comment (peterhaneve) [inline Resculpt/Resculptable.cs:103]: `?.` does not work on destroyed components, as it uses `Object.Equals` which Unity breaks. Either explicitly store and check the `Building`, or use (extension) `GetComponentSafe<T>(this GameObject)`.

File: Resculpt/Resculptable.cs [hunk 3]
```diff
@@ -104,6 +120,14 @@ public sealed class Resculptable : KMonoBehaviour {
 				var button = new KIconButtonMenu.ButtonInfo(icon, text, OnResculpt,
 					PAction.MaxAction, null, null, null, ResculptStrings.RESCULPT_TOOLTIP);
 				Game.Instance?.userMenu?.AddButton(gameObject, button);
+
+				var rotationButton = new KIconButtonMenu.ButtonInfo("action_direction_both",
```
Comment (peterhaneve) [inline Resculpt/Resculptable.cs:124]: Nit: Button is created and immediately destroyed if the object cannot be rotated, consider allocating only inside the `if (Rotatable != null)`

File: Resculpt/Resculptable.cs [hunk 4]
```diff
@@ -104,6 +120,14 @@ public sealed class Resculptable : KMonoBehaviour {
...
+																	"Rotate Art",
+																	new System.Action(this.OnRotateClicked),
+																	Action.BuildMenuKeyO,
+																	tooltipText: "Rotates artwork. {Hotkey}");
```
Comment (peterhaneve) [inline Resculpt/Resculptable.cs:128]: This should be in the `ResculptStrings` so it can be translated

Comment (pether-pg) [reply Resculpt/Resculptable.cs]: moved to ResculptStrings

File: Resculpt/Resculptable.cs (hunk 3 above)
Comment (pether-pg) [reply Resculpt/Resculptable.cs]: moved inside if()

File: Resculpt/Resculptable.cs (hunk 2 above)
Comment (pether-pg) [reply Resculpt/Resculptable.cs]: Building is now stored for the check

File: Resculpt/Resculptable.cs (hunk 1 above)
Comment (pether-pg) [reply Resculpt/Resculptable.cs]: included in if block

**Comment by peterhaneve** [comment]
Closing and accepting #118 instead

**Pull request closed without merging**